compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -3416,12 +3441,21 @@ impl<'a> Parser<'a> {
 
         CondChecker::new(self).visit_expr(&mut cond);
 
-        let (has_let_expr, does_not_have_bin_op) = check_let_expr(&cond);
-        if has_let_expr {
-            if does_not_have_bin_op {
-                // Remove the last feature gating of a `let` expression since it's stable.
-                self.psess.gated_spans.ungate_last(sym::let_chains, cond.span);
+        if has_let_expr(&cond) {
+            // Let chains are allowed in match guards, but only there
```

Comment:
```suggestion
            // Let chains are allowed in match guards, but only there.
```

Also, perhaps you could elaborate on this comment.  It's currently not very clarifying for me.

Reply:
Removed the entire comment. the point was that match guards support let chains, but if you have an `if` inside a block inside a match guard, let chains aren't supported again.